assets/tests/hooks/usePatternsByIdForRoute.test.ts

```diff
@@ -45,76 +49,68 @@ describe("usePatternsByIdForRoute", () => {
       return usePatternsByIdForRoute(null)
     })
     rerender()
-    expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(0)
+    await waitFor(() => expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(0))
   })
 
-  test("doesn't refetch routes on every render", () => {
-    const mockFetchRoutePatterns: jest.Mock =
-      Api.fetchRoutePatterns as jest.Mock
-    mockFetchRoutePatterns.mockReturnValueOnce(
-      instantPromise([routePatternFactory.buildList(2)])
-    )
-    const { rerender } = renderHook(() => {
-      return usePatternsByIdForRoute("66")
+  test("doesn't refetch routes on every render", async () => {
+    jest
+      .mocked(Api.fetchRoutePatterns)
+      .mockResolvedValueOnce(routePatternFactory.buildList(2))
+
+    const { rerender, result } = renderHook(usePatternsByIdForRoute, {
+      initialProps: "66",
     })
-    rerender()
-    expect(mockFetchRoutePatterns).toHaveBeenCalledTimes(1)
+
+    rerender("66")
+
+    await waitFor(() => expect(result.current).not.toBeNull())
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenCalledTimes(1)
   })
 
-  test("refetches when routeId changes", () => {
-    const mockFetchRoutePatterns: jest.Mock =
-      Api.fetchRoutePatterns as jest.Mock
-    mockFetchRoutePatterns.mockReturnValueOnce(
-      instantPromise([routePatternFactory.buildList(2)])
-    )
-    const { rerender } = renderHook(
-      (id) => {
-        return usePatternsByIdForRoute(id)
-      },
-      { initialProps: "66" }
-    )
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("66")
+  test("refetches when routeId changes", async () => {
+    const { rerender } = renderHook(usePatternsByIdForRoute, {
+      initialProps: "66",
+    })
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("66")
+
     rerender("39")
-    expect(mockFetchRoutePatterns).toHaveBeenLastCalledWith("39")
+
+    expect(jest.mocked(Api.fetchRoutePatterns)).toHaveBeenLastCalledWith("39")
   })
 
   test("Keeps most recently requested route patterns if previous resolves finishes later", async () => {
-    const patternFor66 = routePatternFactory.build({ routeId: "66" })
-    // const patternsFor66ById = { [patternFor66.id]: patternFor66 }
+    const { promise, resolve } =
```

Comment:
Non-blocking, but calling for a follow-up: It would be nice if this test were clearer about how it tests what it claims to test. 

I think making some of the variable names clearer, and also putting in some explicit literals, like `"39"` and `"66"` would make the use case of this test easier to follow.

Reply:
how do you feel about dropping it entirely and relying on the underlying test from https://github.com/mbta/skate/blob/main/assets/tests/hooks/useApiCall.test.ts#L100-L124 ?

Reply:
Happy to keep discussing, I'm going to merge but we can do something as a follow up.